File: app/src/main/java/com/example/newsapp/MainActivity.java
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private Spinner spinnerRegion, spinnerTopic;
    private Button buttonGetNews;
    private String selectedRegion = "us"; // Default region
    private String selectedTopic = "general"; // Default topic

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        spinnerRegion = findViewById(R.id.spinnerRegion);
        spinnerTopic = findViewById(R.id.spinnerTopic);
        buttonGetNews = findViewById(R.id.buttonGetNews);

        setupSpinners();
        setupButtonListener();
    }

    private void setupSpinners() {
        ArrayAdapter<CharSequence> regionAdapter = ArrayAdapter.createFromResource(this,
                R.array.regions, android.R.layout.simple_spinner_item);
        regionAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerRegion.setAdapter(regionAdapter);

        spinnerRegion.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                selectedRegion = parent.getItemAtPosition(position).toString().toLowerCase();
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {}
        });

        ArrayAdapter<CharSequence> topicAdapter = ArrayAdapter.createFromResource(this,
                R.array.topics, android.R.layout.simple_spinner_item);
        topicAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerTopic.setAdapter(topicAdapter);

        spinnerTopic.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                selectedTopic = parent.getItemAtPosition(position).toString().toLowerCase();
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {}
        });
    }

    private void setupButtonListener() {
        buttonGetNews.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, NewsActivity.class);
            intent.putExtra("region", selectedRegion);
            intent.putExtra("topic", selectedTopic);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/newsapp/MainActivity.kt
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var spinnerRegion: Spinner
    private lateinit var spinnerTopic: Spinner
    private lateinit var buttonGetNews: Button
    private var selectedRegion: String = "us" // Default region
    private var selectedTopic: String = "general" // Default topic

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        spinnerRegion = findViewById(R.id.spinnerRegion)
        spinnerTopic = findViewById(R.id.spinnerTopic)
        buttonGetNews = findViewById(R.id.buttonGetNews)

        setupSpinners()
        setupButtonListener()
    }

    private fun setupSpinners() {
        val regionAdapter = ArrayAdapter.createFromResource(this,
            R.array.regions, android.R.layout.simple_spinner_item)
        regionAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinnerRegion.adapter = regionAdapter

        spinnerRegion.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
                selectedRegion = parent.getItemAtPosition(position).toString().lowercase()
            }

            override fun onNothingSelected(parent: AdapterView<*>) {}
        }

        val topicAdapter = ArrayAdapter.createFromResource(this,
            R.array.topics, android.R.layout.simple_spinner_item)
        topicAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinnerTopic.adapter = topicAdapter

        spinnerTopic.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
                selectedTopic = parent.getItemAtPosition(position).toString().lowercase()
            }

            override fun onNothingSelected(parent: AdapterView<*>) {}
        }
    }

    private fun setupButtonListener() {
        buttonGetNews.setOnClickListener {
            val intent = Intent(this, NewsActivity::class.java)
            intent.putExtra("region", selectedRegion)
            intent.putExtra("topic", selectedTopic)
            startActivity(intent)
        }
    }
}
